src/beacon_service.cpp

```diff
@@ -0,0 +1,242 @@
+  *      and/or other materials provided with the distribution.
+  *   3. Neither the name of STMicroelectronics nor the names of its contributors
+  *      may be used to endorse or promote products derived from this software
+  *      without specific prior written permission.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+#include "beacon_service.h"
+#include "bluenrg_utils.h"
+#include "bluenrg_hal_aci.h"
+#include "hci.h"
+#include "hci_le.h"
+#include "bluenrg_utils.h"
+#include "stm32_bluenrg_ble.h"
+#include "osal.h"
+#include "bluenrg_gap_aci.h"
+#include "bluenrg_gatt_aci.h"
+#include "SPBTLE_RF.h"
+
+/** @addtogroup X-CUBE-BLE1_Applications
+ *  @{
+ */
+
+/** @addtogroup Beacon
+ *  @{
+ */
+
+/** @defgroup SENSOR_SERVICE
+ * @{
+ */
+extern uint8_t bnrg_expansion_board;
+BeaconServiceClass BeaconService;
+
+void Beacon_HCI_Event_CB(void *pckt);
+
+/**
+ * @brief  Begin a BeaconService
+ * @param  addr       : MAC Address
+ * @param  beaconType : Type of Beacon, UID or URL
+ * @param  webURL     : URL for Beacon Type URL
+ * @param  beaconID   : Instace name for UID
+ * @param  nameSpace  : Namespace for UID
+ * @retval BLE_STATUS_SUCCESS if success
+ */
+tBleStatus BeaconServiceClass::begin(uint8_t addr[BDADDR_SIZE], uint8_t beaconType,
+                                     char* webURL, uint8_t *beaconID, uint8_t *nameSpace)
+{
+  uint8_t bdaddr[BDADDR_SIZE];
+  uint16_t service_handle, dev_name_char_handle, appearance_char_handle;
+
+  uint8_t  hwVersion;
+  uint16_t fwVersion;
+
+  int ret;
+
+  if(addr == NULL) {
+    return BLE_STATUS_NULL_PARAM;
+  }
+
+  attach_HCI_CB(Beacon_HCI_Event_CB);
+
+  /* get the BlueNRG HW and FW versions */
+  ret = getBlueNRGVersion(&hwVersion, &fwVersion);
+  if(ret) {
+    PRINTF("Reading Version failed.\n");
+    return ret;
+  }
+
+  /*
+   * Reset BlueNRG again otherwise we won't
+   * be able to change its MAC address.
+   * aci_hal_write_config_data() must be the first
+   * command after reset otherwise it will fail.
+   */
+  BlueNRG_RST();
+
+  if (hwVersion > 0x30) { /* X-NUCLEO-IDB05A1 expansion board is used */
```

Comment:
It seems than some configs depends on the BLE hardware used?  IDB05A1  or IDB04A1 (BLE expansion shield) while it is used for disco. Moreover  `bnrg_expansion_board` clearly identified a shield not the board itself. @cparata do you have some clue/advice on this? It should be fine to clean this?

Reply:
So if we follow the way of the other libraries, we should remove all functions linked to the IDB04A1 module. Do you agree?

Reply:
kept as it is